### DISCUSSION TITLE
Borders (and maybe other symbols) rendering oddly on Windows 10 Powershell

### DISCUSSION THREAD

**Discussion by tconbeer** [Textualize/textual#3375] [Q&A] [answered]
Reporting on behalf of a Harlequin user (@MoeJD), who reported this issue [here](https://github.com/tconbeer/harlequin/issues/237).

Harlequin uses `round` borders almost everywhere. On this user's machine, those look like this:
![image](https://github.com/Textualize/textual/assets/705906/16eb9ffe-2436-45bd-b6dd-2c3ed22f2455)

I cannot reproduce this on my machine (Windows 11).

Here is what we've ruled out so far:
- It's not his terminal or their settings (we have the same Windows Terminal settings and he can repro in the VS Code terminal)
- It's not the font (he first reported this with ProFontWindows Nerd Font Mono, but he can repro using FiraCode, and ProFontWindows works fine on my machine)
- It's not his locale or default encoding (CP-1252, same as mine)

I'm at a dead-end, so I thought I'd ask for some help. I've asked him to try to run `textual diagnose` so we can include his output below.

**Comment by davep** [comment]
Just moving this to discussions as it seems more of a Q&A than an issue, and should be visible longer-term if other folk come looking for the same problem.

**Reply by tconbeer** [reply]
Totally fair. Thanks!

**Reply by davep** [reply]
I may have missed this somewhere: but have you confirmed that they have the same *version* of Windows Terminal as you (ideally the latest version).

I'm also mindful of the fact that the big difference here seems to be Windows 10 vs Windows 11.

**Reply by tconbeer** [reply]
No, that's a good catch. I'm on 1.17.11461.0. (@MoeJD FYI). There are also some options for repairing and resetting the installation that we should try. (This is all at Settings > Apps > Installed Apps > Terminal, at least on my machine).

**Reply by davep** [reply]
One other thing to consider, if it hasn't been so far: the individual profiles can override the general settings, so it may be worth checking the settings for the profile being used (a bit of a stretch, but work keeping in mind just in case).

**Reply by davep** [reply]
> It's not his locale or default encoding (CP-1252, same as mine)

Now this is straying into areas of ignorance for me, but here I'm `utf-8` if I run `[System.Text.Encoding]::Default`; and a quick search online suggests that Windows-1252 is constrained to just 256 characters (which would cut out the box drawing characters, amongst others).

So my likely candidate cause would be this; but you say you're running fine with the same setup. Which then makes me wonder if there's a difference in how Windows Terminal handles that sort of situation on Windows 11, vs 10.

But I'm not a Windows person so this is all pure guesswork and a quick Google.

**Comment by MoeJD** [comment]
My Terminal is also 1.17.11461.0

I am attempting to get `textual diagnose` to work but after `pipx install textual` but I am getting `Error: No such command 'diagnose'`

![image](https://github.com/Textualize/textual/assets/76531825/3e27e917-2998-4e1b-9e45-29767798c6e4)

**Reply by davep** [reply]
@MoeJD What does `textual --version` report?

**Reply by MoeJD** [reply]
![image](https://github.com/Textualize/textual/assets/76531825/52379430-e458-45e5-99c2-4845b4ab6b47)

**Reply by TomJGooding** [reply]
Clearly something odd happening here - Textual is currently on version 0.38.1!

Could you try running `pipx list` and paste the output? It might show some helpful diagnostics.

**Reply by MoeJD** [reply]
![image](https://github.com/Textualize/textual/assets/76531825/1c646bc0-8b6c-49df-afbe-774c539941de)

**Reply by MoeJD** [reply]
Do I need to uninstall textual from somewhere? or run textual-dev using pipx?

**Reply by MoeJD** [reply]
![image](https://github.com/Textualize/textual/assets/76531825/1f0f2bb1-c14b-4661-92d3-f0cd66284edb)

I wonder if this is causing problems, oh the many ways in which python can exist on one's computer =(

**Reply by TomJGooding** [reply]
Possibly, but I thought pipx was designed to avoid these issues! Sadly looks like that command didn't report anything useful.

Just to rule it out, have you tried uninstalling and reinstalling Harlequin?

Out of interest, I see you have `frogmouth` installed. Is that displaying as expected?

**Reply by MoeJD** [reply]
![image](https://github.com/Textualize/textual/assets/76531825/7a33e22d-7574-4909-8d4d-e521b6c18f0c)

How fun, frogmouth has the same issue 😢

**Reply by MoeJD** [reply]
This was from `pipx run frogmouth`

**Reply by TomJGooding** [reply]
Is that your first time running frogmouth, or do you remember it working before?

> This was from `pipx run frogmouth`

It looks like you have it already installed, so it is the same result just running `frogmouth`?

**Reply by MoeJD** [reply]
After uninstalling and reinstalling using pipx I am still getting the same border errors.

**Reply by MoeJD** [reply]
I thank you for your grace and patience in helping me with this issue, Neither `harelquin` or `frogmouth` work from my shell directly for some reason, I can get them to launch using `pipx run {command}` but with the border errors.

![image](https://github.com/Textualize/textual/assets/76531825/1753ce8f-8752-4e7e-90fc-8f9730fef4a6)

**Reply by MoeJD** [reply]
> Is that your first time running frogmouth, or do you remember it working before?

I have not gotten either of the apps to work without the border errors before.

**Reply by davep** [reply]
I would suspect the issue isn't particular to Textual applications. 

I sense it's worth exploring the utf-8 aspect I mentioned earlier.

**Reply by MoeJD** [reply]
I would like to explore those options as well, without being able to get `textual diagnose` to work do you have any other suggestions on what I can provide or where I can get started to look into that?

or if you have any ideas on how to get `textual diagnose` to run.

Very much appreciate all of your help in this!

**Reply by TomJGooding** [reply]
Fair point @davep. It just seems odd that the Powerline prompt is fine, but not the basic blocks used in Textual borders.

**Reply by MoeJD** [reply]
The `textual` demo looks to be working just fine

![image](https://github.com/Textualize/textual/assets/76531825/ebd7217f-c9ca-4146-bb71-ef6554045db9)

**Reply by TomJGooding** [reply]
I think that rules out the system/terminal being the issue then. So how did you run the textual demo?

**Reply by MoeJD** [reply]
`python3 -m textual` for this screenshot

**Reply by TomJGooding** [reply]
Sorry I'm getting a bit lost what's been installed with pipx vs simply pip. I think you might have textual installed globally via pip (not pipx), is that correct? If that's the case, then all this would suggest that pipx might be the problem. If nothng else, there is clearly something wrong with your installion if you can't actually invoke the programs you've installed.

**Reply by tconbeer** [reply, marked as answer]
Thanks for your help, Tom! I agree -- seems like the Python/pip/pipx installation is the problem. @MoeJD if I were you I would probably nuke everything and start over.

**Comment by MoeJD** [comment]
# Textual Diagnostics

## Versions

| Name    | Value  |
|---------|--------|
| Textual | 0.38.1 |
| Rich    | 13.5.3 |

## Python

| Name           | Value                                |
|----------------|--------------------------------------|
| Version        | 3.8.5                                |
| Implementation | CPython                              |
| Compiler       | MSC v.1924 64 bit (AMD64)            |
| Executable     | c:\program files\python38\python.exe |

## Operating System

| Name    | Value      |
|---------|------------|
| System  | Windows    |
| Release | 10         |
| Version | 10.0.19041 |

## Terminal

| Name                 | Value            |
|----------------------|------------------|
| Terminal Application | Windows Terminal |
| TERM                 | *Not set*        |
| COLORTERM            | *Not set*        |
| FORCE_COLOR          | *Not set*        |
| NO_COLOR             | *Not set*        |

## Rich Console options

| Name           | Value                |
|----------------|----------------------|
| size           | width=252, height=67 |
| legacy_windows | False                |
| min_width      | 1                    |
| max_width      | 252                  |
| is_terminal    | True                 |
| encoding       | utf-8                |
| max_height     | 67                   |
| justify        | None                 |
| overflow       | None                 |
| no_wrap        | False                |
| highlight      | None                 |
| markup         | None                 |
| height         | None                 |
